test/functional/tests/lazy_writes/cleaning_policy/test_alru.py

```diff
@@ -118,3 +124,201 @@ def prepare():
+        core_dev.create_partitions([Size(10, Unit.GiB)])
+
+    with TestRun.step("Disable udev"):
+        Udev.disable()
+
+    with TestRun.step("Start cache and add core"):
+        cache = casadm.start_cache(cache_dev.partitions[0], force=True, cache_mode=CacheMode.WB)
+        core = cache.add_core(core_dev.partitions[0])
+
+    with TestRun.step("Set ALRU and disable sequential cutoff"):
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+        cache.set_cleaning_policy(CleaningPolicy.alru)
+
+    with TestRun.step(
+        "Configure ALRU: dirty threshold=50%, inertia={inertia}, default wakeup, max staleness_time"
+    ):
+        cache.set_params_alru(
+            FlushParametersAlru(
+                staleness_time=Time(seconds=3600),
+                dirty_ratio_threshold=target_dirty_ratio_threshold,
+                dirty_ratio_inertia=inertia,
+            )
+        )
+
+    with TestRun.step("Start background fio (4 GiB randwrite)"):
+        (
+            Fio()
+            .create_command()
+            .io_engine(IoEngine.libaio)
+            .size(Size(4, Unit.GiB))
+            .block_size(Size(4, Unit.KiB))
+            .target(core)
+            .direct()
+            .read_write(ReadWrite.randwrite)
+            .run_in_background()
+        )
+
+    with TestRun.step("Wait until cache dirty >= 90%, then stop workload"):
+        start_ts = time.time()
+        while True:
+            time.sleep(2)
+            dirty_pct = cache.get_statistics(percentage_val=True).usage_stats.dirty
+            if dirty_pct >= 90:
+                break
+            if time.time() - start_ts > 5 * 60:
+                TestRun.LOGGER.exception(
+                    f"Exception: Cache dirty level did not reach 90% within 5 minutes "
+                    f"(current dirty={dirty_pct}%). Aborting test."
+                )
+
+    with TestRun.step("Stop the IO"):
+        kill_all_io()
+
+    with TestRun.step(
+        "Observe cleaning: ensure monotonic decrease and stop at (threshold - inertia)"
+    ):
+        # Initial dirty after IO stop
+        last_dirty = cache.get_statistics().usage_stats.dirty
+
+        not_decreasing = 0
+        while last_dirty > target_cleaning_threshold:
+            time.sleep(5)
+            dirty_now = cache.get_statistics().usage_stats.dirty
+
+            if dirty_now < last_dirty:
+                not_decreasing = 0
+            else:
+                not_decreasing += 1
+                if not_decreasing >= 20:
+                    TestRun.fail(
+                        f"Exception: Dirty amount not decreasing for 20 consecutive checks "
+                        f"(stuck at {dirty_now}, target stop {target_cleaning_threshold})."
+                    )
+
+            last_dirty = dirty_now
+
+    with TestRun.step(
+        "Once the dirty cache lines counter reached (threshold - inertia) level, "
+        "no further cleaning should take place"
```

Comment:
This sounds more like a description than a step
How about
```suggestion
        "Verify if cleaning stop after reaching target cleaning threshold"
```